### ISSUE TITLE
Crashes If a Scope hasn't been Properly Closed

### ISSUE THREAD

**Issue by sathukorale** [sathukorale/C-PlantUML-Class-Diagram-Generator#3]
If a detected scope hasn't been properly closed (reason yet to be determined) it can cause the PlantUML code generation logic to fail. The exception at [Class.cs:862](https://github.com/sathukorale/C-PlantUML-Class-Diagram-Generator/blob/34a08d5c78939306f14d10360319bfc08daf1616/PlantUMLCodeGeneratorGUI/Class.cs#L862) will be thrown if this issue occurs.

The following code segment, and to be precise the lines from 850 to 859 causes this issue.
https://github.com/sathukorale/C-PlantUML-Class-Diagram-Generator/blob/34a08d5c78939306f14d10360319bfc08daf1616/PlantUMLCodeGeneratorGUI/Class.cs#L844-L863

For example, if the code segments has been running for awhile and if the `numberOfClosingBrackets` is still not `0`, and if there are no more `{` or `}` this will make 
```
indexOfOpeningBracket = -1;
indexOfClosingBracket = -1;
```
so this will make the code go through line `857` which will make 
```
indexToCheckFrom = -1;
```

which essentially means we are starting all over again, and if not handled will make the code run for eternity.

**Comment by sathukorale** [comment]
Turns out the original issue of having not closed classes was caused by the comment removal logic. For example take this class
```c++
class SomeClass
{
 public:
  /* This is some example comment
   */// And this is another comment. 
  SomeClass() {  }
};
```
The comment removal process first removed single line comments that start with `//` and as such it detects the first such occurrence at the line,
```
   */// And this is another comment. 
```
However instead of selecting the second set of `//` it selects the first `//` which contains a `/` that is part of the `*/` closing multi-line comment. So since this is a single line comment, the logic removes the entire line after that occurrence. So what we are left with is,
```c++
class SomeClass
{
 public:
  /* This is some example comment
   *
  SomeClass() {  }
};
```
And when the multi-line comment removal process starts, it doesn't see the end to comment, which in turn removes the entire content after the start of the comment. So what we are left with is,
```c++
class SomeClass
{
 public:
```

**Comment by sathukorale** [comment]
This issue was fixed by the commit, 816fd5beae5bdfb0a15bf30664f81e041dfa1c89. However the logic to detect this issue (just in case if the class is not actually closed) will be left in place.
